### ISSUE TITLE
Twig as a First-Class Citizen and final ControllerTrait::render()

### ISSUE THREAD

**Issue by umpirsky** [symfony/symfony#26255]
| Q                | A
| ---------------- | -----
| Bug report?      | no
| Feature request? | yes
| BC Break report? | yes
| RFC?             | yes
| Symfony version  | 3.4.4

As described in http://symfony.com/blog/new-in-symfony-2-7-twig-as-a-first-class-citizen#what-about-performance, there is a performance penalty when using templating sub-system.

[ControllerTrait::render()](https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/FrameworkBundle/Controller/ControllerTrait.php#L219-L235) tries to render with `templating` first, but one can optimize controllers by overriding it in base controller like:

```php
public function render($view, array $parameters = [], Response $response = null)
{
    if (null === $response) {
        $response = new Response();
    }

    $response->setContent($this->get('twig')->render($view, $parameters));

    return $response;
}
```

The method `ControllerTrait::render()` is considered final [since version 3.4](https://github.com/symfony/symfony/pull/24407), so it cannot be overridden any more, and this optimization is no longer possible.

One possibility can be:
```yaml
templating:
        enabled: false
```
but that can cause issues with some third party code that rely on `templating` service.

So basically what I am trying to accomplish is to keep this optimization possible for controllers.

One idea was to try `twig` first in [ControllerTrait::render()](https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/FrameworkBundle/Controller/ControllerTrait.php#L219-L235), but that would probably break in some cases.

So, is there a workaround I am not aware of, or is there something we can do to allow this kind of optimizations in future versions?

**Comment by umpirsky** [comment]
@Tobion Thanks, I thought that at first, but then I read http://symfony.com/blog/new-in-symfony-2-7-twig-as-a-first-class-citizen#what-about-performance. Is it true that "optimizing one `Container::has` call" drops memory consumption by 30%? If true, I really need to optimize it.

**Comment by umpirsky** [comment]
That memory consumption by 30% from http://symfony.com/blog/new-in-symfony-2-7-twig-as-a-first-class-citizen#what-about-performance in my case is not even close. I just profiled with and without optimization and the difference is minor and not worth caring about:

![screenshot from 2018-02-22 11-22-51](https://user-images.githubusercontent.com/208957/36556323-0af8dec6-1805-11e8-91fd-2bb98a2c560b.png)

Thanks!
